specs/_features/maxeb_increase/capella.py

```diff
@@ -1776,6 +1790,49 @@ def process_execution_layer_withdraw_request(
     ))
 
 
+def is_consolidated(validator: Validator) -> bool:
+   validator.consolidated_to != UNSET_CONSOLIDATED_TO
+
+
+def process_consolidation(state: BeaconState, consolidation: Consolidation) -> None:
+    target_validator = state[consolidation.target_index]
+    source_validator = state[consolidation.source_index]
+
+    assert is_active_validator(target_validator) and not target_validator.slashed and not is_consolidated(target_validator)
+    assert not source_validator.slashed and not is_consolidated(source_validator)
+
+    # verify source withdrawal credentials, which have authority over validating keys
+    assert source_validator.withdrawal_credentials[:1] == ETH1_ADDRESS_WITHDRAWAL_PREFIX
+    assert source_validator.withdrawal_credentials[12:] == consolidation.source_address
+
+    # target validator accepts consolidation operation onto itself
+    # Fork-agnostic domain since address changes are valid across forks
+    domain = compute_domain(DOMAIN_CONSOLIDATION, genesis_validators_root=state.genesis_validators_root)
+    signing_root = compute_signing_root(source_validator.pubkey, domain)
+    assert bls.Verify(target_validator.pubkey, signing_root, consolidation.target_signature)
```

Comment:
Target validator authorizes the consolidation. Since the target validator is not liable for source's actions this check may not be strictly necessary but sounds sensible of a first approach. Target validator signs over the public key of the source validator, not the index to prevent replay if source's validator index is ever re-used. There may be some other replay edge cases not covered by the current signed message